Gopkg.lock

```diff
@@ -226,20 +227,20 @@
   revision = "e979a0b10eebe748549c702a25e997c556349da6"
 
 [[projects]]
-  digest = "1:56a1f3949ebb7fa22fa6b4e4ac0fe0f77cc4faee5b57413e6fa9199a8458faf1"
+  digest = "1:a1578f7323eca2b88021fdc9a79a99833d40b12c32a5ea4f284e2fad19ea2657"
   name = "github.com/google/uuid"
   packages = ["."]
   pruneopts = "NUT"
-  revision = "9b3b1e0f5f99ae461456d768e7d301a7acdaa2d8"
-  version = "v1.1.0"
+  revision = "d460ce9f8df2e77fb1ba55ca87fafed96c607494"
+  version = "v1.0.0"
```

Comment:
It looks like this and most of the other version changes in Gopkg.lock are reverting the dependency bumps from #155 . A merge issue? Or `update-deps.sh` reverting the changes to Gopkg.lock from that other PR?

Reply:
Interesting, good catch @bbrowning. Here's what I did to get this result:

```
git merge master
git reset Gopkg.lock && git co Gopkg.lock
hack/update-deps.sh
```

This is not correct since `Gopkg.lock` has information about the specific revisions pinned that is not duplicated elsewhere. I redid the merge like this:

```
git merge -X theirs master
hack/update-deps.sh
```

The `-X theirs` option tells the merge strategy to resolve conflicts by choosing the remote version  ("theirs") over the local version ("ours").

In the future we can avoid this by always updating `Gopkg.toml` when we bump dependencies. For example, #155 could have added this constraint to avoid a future downgrade:

```toml
[[constraint]]
  name = "github.com/google/uuid"
  version = "1.1.0"
```
See https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md for more details on the `version` property.

/cc @vdemeester

Reply:
> I redid the merge like this

Forgot to mention that when I did this, I got the following error from `hack/update-deps.sh`:
```
cannot Stat: stat /usr/local/google/home/grantrodgers/go/src/github.com/knative/eventing-sources/vendor/github.com/knative/eventing/test/test_images/k8sevents/kodata/VENDOR-LICENSE: no such file or directory
```

I think this quick hack has become more trouble that its worth at this point and should probably be abandoned in favor of a different approach. Sources don't actually need eventing primitives to run e2e tests - they can be tested with direct sinks instead.

Reply:
@grantr Adding a constraint for each dependency seems counter-productive (although having something like `^1.2.3` would make sure we don't update to major version…). I think a rebase would have been easier to handle (for the conflict).

```
git fetch -p --all
git rebase origin/master # or upstream/master
# on the rebase that fail
git checkout origin/master Gopkg.lock # or upstream/master
git rebase --continue
```

> This is not correct since Gopkg.lock has information about the specific revisions pinned that is not duplicated elsewhere. I redid the merge like this:

Looking at the [commit](https://github.com/knative/eventing-sources/pull/121/commits/917c515d10840bceb77da434145ac609dcbda1be) that update dependencies, it's safe to take the upstream master branch version as there is no actual bump of version (and the `packages` and `digest` information will be re-computed when doing `dep ensure`)

That said, I don't understand why the merge did revert some versions and not others :thinking:

Reply:
> I'd like to understand the objection here. Why would this be counter-productive? Note that for projects using go modules, bumping a version would always require a constraint due to MVS.

Right, but you will "more or less" do a `go get -u the/dep` for "safe" bumps and `go get the/dep@version` for other bump (aka bigger one).

But you have a point on the rest, to be really safe, a constraint with `^` (which is default indeed :stuck_out_tongue_closed_eyes:) seems ok, it's just a bit of a pain to write/maintain :angel:. It's gonna be easier with go modules (as you will be able to "only" use the command-line :yum:).

> I prefer to avoid rebasing PRs that have already been reviewed, but to each their own :)

Indeed :angel: wasn't trying to convince or anything… (my workflow almost never involve a `merge` :sweat_smile:)